src/Application/Response/UserCreationResponse.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Application\Response;
+
+class UserCreationResponse implements \JsonSerializable
```

Comment:
Usually you go with the verb for the name of requests and responses, just like you did with the use case. So a better name with this would be CreateUserResponse. Also is so much easier to know who's the producer of this response if you use the same naming convention for controllers, requests, handlers, use cases and responses.